web/src/routes/(user)/+layout.svelte

```diff
@@ -22,8 +22,27 @@
       await fileUploadHandler(filesArray, albumId);
     }
   };
+
+  const pasteHandler = async ({ clipboardData }: ClipboardEvent) => {
+    const files = clipboardData?.files;
+    if (!files) {
+      return;
+    }
+
+    const filesArray: File[] = Array.from<File>(files);
+    albumId = ($page.route.id === '/(user)/albums/[albumId]' || undefined) && $page.params.albumId;
+
+    const isShare = $page.route.id === '/(user)/share/[key]' || undefined;
+    if (isShare) {
+      dragAndDropFilesStore.set({ isDragging: true, files: filesArray });
+    } else {
+      await fileUploadHandler(filesArray, albumId);
+    }
```

Comment:
Maybe it would be worthwhile to extract that logic to a private function as it's the exact same the `dropHandler` uses. Avoid some duplication etc